source/Halibut/Transport/SecureClient.cs

```diff
@@ -137,21 +138,20 @@ void HandleError(Exception lastError, bool retryAllowed)
 
             lastError = lastError.UnpackFromContainers();
 
+            var innermost = lastError;
+            while (innermost.InnerException != null)
+                innermost = innermost.InnerException;
+            
+            if (innermost is SocketException se && !retryAllowed)
+                if (se.SocketErrorCode == SocketError.ConnectionAborted || se.SocketErrorCode == SocketError.ConnectionReset)
+                    throw new HalibutClientException($"The server {ServiceEndpoint.BaseUri} aborted the connection before it was fully established. This usually means that the server rejected the certificate that we provided. We provided a certificate with a thumbprint of '{clientCertificate.Thumbprint}'.");
+
```

Comment:
This cleans up the error message that gets returned without the extra guff that was previously added